### PR TITLE
61140 improve save parallelism

### PR DESCRIPTION
**Pull request by mayya-sharipova**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -270,11 +364,10 @@ conf.set("jsonstore.rdd.schemaSampleSize", -1)
 sc = SparkContext(conf=conf)
 sqlContext = SQLContext(sc)
 ```
-
 For a local setting applied to a single RDD only, use:
 
-```
-sqlContext.sql("CREATE TEMPORARY TABLE schema-test USING com.cloudant.spark.CloudantRP OPTIONS ( schemaSampleSize '10',database 'schema-test')")
+``` python
+sqlContext.sql("CREATE TEMPORARY TABLE schema-test USING com.cloudant.spark OPTIONS ( schemaSampleSize '10',database 'schema-test')")
```
Comment (ykye) [inline README.md:370]: `schemaSampleSize` should change to `jsonstore.rdd.schemaSampleSize`

File: examples/python/CloudantDF.py [hunk 2]
```diff
@@ -23,8 +23,9 @@
 # conf.set("cloudant.protocol","http")
 conf.set("cloudant.host","ACCOUNT.cloudant.com")
 conf.set("cloudant.username", "USERNAME")
-conf.set("cloudant.password","PASSWORD")
+conf.set("cloudant.password", "PASSWORD")
 conf.set("jsonstore.rdd.maxInPartition",1000)
+conf.set("schemaSampleSize", 10)
```
Comment (ykye) [inline examples/python/CloudantDF.py:28]: This should be `jsonstore.rdd.schemaSampleSize`

File: examples/python/CloudantDF.py [hunk 3]
```diff
@@ -45,7 +46,7 @@
 
 df.filter(df.airportName >= 'Moscow').select("_id",'airportName').show()
 df.filter(df._id >= 'CAA').select("_id",'airportName').show()
-df.filter(df._id >= 'CAA').select("_id",'airportName').save("airportcodemapping_df", "com.cloudant.spark")
+df.filter(df._id >= 'CAA').select("_id",'airportName').save("airportcodemapping_df", "com.cloudant.spark", bulkSize = "100")
```
Comment (ykye) [inline examples/python/CloudantDF.py:49]: should be `jsonstore.rdd.bulkSize`

File: examples/python/CloudantDFOption.py [hunk 4]
```diff
@@ -30,7 +29,7 @@
 cloudant_username = "USERNAME"
 cloudant_password = "PASSWORD"
 
-df = sqlContext.read.format("com.cloudant.spark").option("cloudant.host",cloudant_host).option("cloudant.username",cloudant_username).option("cloudant.password",cloudant_password).load("n_airportcodemapping")
+df = sqlContext.read.format("com.cloudant.spark").option("cloudant.host",cloudant_host).option("cloudant.username",cloudant_username).option("cloudant.password",cloudant_password).option("schemaSampleSize", "10").load("n_airportcodemapping")
```
Comment (ykye) [inline examples/python/CloudantDFOption.py:32]: This should be `jsonstore.rdd.schemaSampleSize`

File: examples/python/CloudantDFOption.py [hunk 5]
```diff
@@ -44,7 +43,7 @@
 df.printSchema()
 
 df.filter(df._id >= 'CAA').select("_id",'airportName').show()
-df.filter(df._id >= 'CAA').select("_id",'airportName').write.format("com.cloudant.spark").option("cloudant.host",cloudant_host).option("cloudant.username",cloudant_username).option("cloudant.password",cloudant_password).save("airportcodemapping_df")
+df.filter(df._id >= 'CAA').select("_id",'airportName').write.format("com.cloudant.spark").option("cloudant.host",cloudant_host).option("cloudant.username",cloudant_username).option("cloudant.password",cloudant_password).option("bulkSize","100").save("airportcodemapping_df")
```
Comment (ykye) [inline examples/python/CloudantDFOption.py:46]: should be `jsonstore.rdd.bulkSize`

File: test/test-scripts/cloudantapp/IndexOption.py [hunk 6]
```diff
@@ -51,25 +51,3 @@ def verifyIndexOption():
 print ('About to test com.cloudant.spark for n_flight with index')
 sqlContext.sql(" CREATE TEMPORARY TABLE flightTable1 USING com.cloudant.spark OPTIONS ( database 'n_flight', index '_design/view/_search/n_flights')")
 verifyIndexOption()
```
Comment (ykye) [inline test/test-scripts/cloudantapp/IndexOption.py:53]: Maybe need to add `sc.stop()`

File: test/test-scripts/schema/schema_sample_size_001.py [hunk 7]
```diff
@@ -25,8 +25,8 @@
 sc = SparkContext(conf=conf)
 sqlContext = SQLContext(sc)
 
-print ('About to test com.cloudant.spark.CloudantRP for n_customer with setting schemaSampleSize to 0')
-sqlContext.sql(" CREATE TEMPORARY TABLE customerTable USING com.cloudant.spark.CloudantRP OPTIONS ( schemaSampleSize '0',database 'n_customer')")
+print ('About to test com.cloudant.spark for n_customer with setting schemaSampleSize to 0')
+sqlContext.sql(" CREATE TEMPORARY TABLE customerTable USING com.cloudant.spark OPTIONS ( schemaSampleSize '0',database 'n_customer')")
```
Comment (ykye) [inline test/test-scripts/schema/schema_sample_size_001.py:29]: should be `jsonstore.rdd.schemaSampleSize`

File: test/test-scripts/schema/schema_sample_size_002.py [hunk 8]
```diff
@@ -25,8 +25,8 @@
 sc = SparkContext(conf=conf)
 sqlContext = SQLContext(sc)
 
-print ('About to test com.cloudant.spark.CloudantPrunedFilteredRP for n_customer with setting schemaSampleSize to 0')
-sqlContext.sql(" CREATE TEMPORARY TABLE customerTable USING com.cloudant.spark.CloudantPrunedFilteredRP OPTIONS ( schemaSampleSize '0',database 'n_customer')")
+print ('About to test com.cloudant.spark for n_customer with setting schemaSampleSize to 0')
+sqlContext.sql(" CREATE TEMPORARY TABLE customerTable USING com.cloudant.spark OPTIONS ( schemaSampleSize '0',database 'n_customer')")
```
Comment (ykye) [inline test/test-scripts/schema/schema_sample_size_002.py:29]: should be `jsonstore.rdd.schemaSampleSize`

**Comment by mayya-sharipova** [comment]
Closed for https://github.com/cloudant-labs/spark-cloudant/pull/56

**Pull request closed without merging**